Sources/BugsnagPerformance/Public/BugsnagPerformanceSpan.mm

```diff
@@ -175,6 +178,22 @@ - (BOOL)isValid {
 }
 
 - (void)setAttribute:(NSString *)attributeName withValue:(id)value {
+    @synchronized (self) {
+        if(value != nil &&
+           self.attributes[attributeName] == nil &&
+           self.attributes.count >= self.attributeCountLimit) {
+            BSGLogError(@"Called setAttribute on span %llu (%@), but span is over the limit of %lu attributes", self.spanId, self.name, (unsigned long)self.attributeCountLimit);
+            return;
+        }
+        if(attributeName.length > MAX_ATTRIBUTE_KEY_LENGTH) {
```

Comment:
Isn't this also being validated on serialisation? That was the intent.

Reply:
If we validate it here, we'll never get an oversized key when serializing. The length is already known at this point, so it seemed better to just nip it in the bud rather than carry around an attribute that will ultimately be discarded anyway.

Reply:
Understood - but maybe worth removing the later check? (and mentioning to the rest of the team for other implementations?)

Reply:
Actually, it looks like I did put it in the encoder as well, so we're fully covered in case we screw up as well...

Reply:
I've updated the messaging to what we discussed earlier.

Reply:
I'll remove the late check as well